### PR TITLE
CICD: Update AppImage Workflow to Be Based on ubuntu 20.04 Image.

### PR DESCRIPTION
**Pull request by mikeller**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x] Build system change
- [ ] Documentation change
- [ ] Language translation

### Changes made:

Update the AppImage workflow to be based on ubuntu 20.04 in order to support the updated requirement of GitHub actions to run on node 20.

### CODE REVIEW

**Comment by mikeller** [comment]
@probonopd: We seem to be a bit stuck between a rock and a hard place with this:
GitHub actions seem to now require at least node 20, which depends on glibc >= 2.25:

```
 ##[debug]Running JavaScript Action with default external tool: node20
43
/__e/node20/bin/node: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /__e/node20/bin/node)
44
/__e/node20/bin/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)
45
/__e/node20/bin/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by /__e/node20/bin/node)
46
##[debug]Node Action run completed with exit code 1
```

On the other hand, the AppImage build scripts seem to expect a glibc <= 2.23:

```
ERROR: The host system is too new.
49
Please run on a system with a glibc version no newer than what comes with the oldest
50
currently still-supported mainstream distribution (xenial), which is glibc 2.23.
51
```

What is the correct way to to make the AppImage build run on a distribution that is supported by GitHub workflows?

**Comment by probonopd** [comment]
The most recent builds of linuxdeployqt should work on Ubuntu Focal Fossa, which has glibc 2.31. You can get it [here](https://github.com/probonopd/linuxdeployqt/releases/tag/continuous). Let me know if you experience any issues with it.

**Comment by mikeller** [comment]
@probonopd: Thank you. Do you have any idea when we can expect the next released version of the tool? In the interest of getting repeatable builds and minimising the risk of things breaking without change I am keen to base our builds on fixed released versions.

File: .github/workflows/linux-ubuntu-20.04-qt5-appimage.yml [hunk 1]
```diff
@@ -112,7 +96,10 @@ jobs:
         cp /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1 appdir/usr/lib/
 
         # get the linuxdeployqt tool and run it to collect the libraries
-        curl -L -O "https://github.com/probonopd/linuxdeployqt/releases/download/7/linuxdeployqt-7-x86_64.AppImage"
+        #
+        # ideally we should switch to using an actually released version of
+        # the tool, or alternatively cache a fixed version on our side
+        curl -L -O "https://github.com/probonopd/linuxdeployqt/releases/download/continuos/linuxdeployqt-continuos-x86_64.AppImage"
```
Comment (glance-) [inline .github/workflows/linux-ubuntu-20.04-qt5-appimage.yml]: Spelling, continuos vs. continuous

**Comment by mikeller** [comment]
@dirkhh: Not sure what to do here - GitHub has dropped support for MacOS 11, and Xcode 11 along with it. I can't quite get it to build on MacOS 12 / Xcode 13, as it seems to want to build the simulator for arm64, but then cannot build the dependency. This will need some debugging by somebody who can run these versions locally.
Or we could just disable the iOS simulator CICD job for now - this isn't a job that creates an artifact that is added to the release.

**Comment by dirkhh** [comment]
I have it working with macOS 12 locally. 
I need to figure out what's different. Morning stands out in the logs, TBH.
Let's disable for now while I come up with a solution

**Comment by mikeller** [comment]
Thanks @dirkhh. I'll merge this to get our release pipelines unblocked.

**Pull request merged**